ksqldb-engine/src/main/java/io/confluent/ksql/internal/MetricsTagUtils.java

```diff
@@ -27,7 +27,30 @@ private MetricsTagUtils() {
   public static final String KSQL_TOPIC_TAG = "topic";
   public static final String KSQL_QUERY_ID_TAG = "query-id";
 
-  public static final Pattern NAMED_TOPOLOGY_PATTERN = Pattern.compile("(.*?)__\\d*_\\d*");
-  public static final Pattern QUERY_ID_PATTERN =
-      Pattern.compile("(?<=query_|transient_)(.*?)(?=-)");
+  public static final Pattern SHARED_RUNTIME_THREAD_PATTERN = Pattern.compile("(.*?)__\\d*_\\d*");
+
+  /*
+   For non-shared runtimes, the thread id will look something like this:
+
+   _confluent-ksql-pksqlc-d1m0zquery_ +                  // thread id prefix
+   CSAS_TEST_COPY-STREAM_1_23 +                          // query id
+   -3d62ddb9-d520-4cb3-9c23-968f8e61e201-StreamThread-1  // thread id suffix
+   */
+  public static final String UUID_PATTERN =
+      "-[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}";
+  public static final String THREAD_ID_SUFFIX_PATTERN = UUID_PATTERN + "-StreamThread-\\d";
+  public static final String THREAD_ID_PREFIX_PATTERN = "query_|transient_";
```

Comment:
You asked the same question  when you reviewed the original P R that added this class --  IIRC it was (only) the internal topic names that used the hyphen instead of the underscore, whereas the thread id always has an underscore here. I did double check on that last time so I'm pretty confident now that this is correct

Reply:
ok actually now I remember, the reason we use `_` here is just because this regex is only used for non-shared  runtimes. So ignore the above lol